### PR TITLE
VACMS-10785 connect homepage prototype to backend

### PR DESCRIPTION
**Pull request by apisandipas**
## Description
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10785
Connects the new homepage prototype template to the Drupal backend via GraphQL. This PR modified the templates and integrated the queries into the home page build script.

## Testing done
Manual testing completed locally.

## Acceptance Criteria
- [ ] Benefit Promo block content displays title, description, and CTA with a link using a new clearly labeled content block type in Drupal.
- [ ] VA News promo content displays link, description, CTA with link and photo using Drupal content. 
- [ ] Popular on VA.gov link menu references heading, links, link labels, and their ordering from Drupal. 
- [ ] The Other search tools link menu references heading, links, link labels, and their ordering from Drupal.  

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by ryguyk — CHANGES_REQUESTED** [review]
This looks really good. I requested a couple very small changes.

File: src/site/stages/build/drupal/graphql/homePage.graphql.js [hunk 1]
```diff
@@ -81,6 +87,96 @@ const query = `
...
+      url {
+        path
+      }
+      links {
+        label
+        url {
+          path
+        }
+      }
+    }
```
[the remark this answers is not in the document]
Comment (apisandipas) [reply src/site/stages/build/drupal/graphql/homePage.graphql.js]: Negative on the nested links requirement. I merely copied the queries as they were supplied. I will remove them.

**Comment by jilladams** [comment]
Updated to point to Integration branch (per [discussion with Ryan](https://dsva.slack.com/archives/C52CL1PKQ/p1666376331770019)) and removed Do Not Merge label. @apisandipas if nothing else is blocking review, feel free to un-draft it and get this merged to integration.

**Review by ryguyk — APPROVED** [review]
Looks good. Thanks for making those few changes.

**Comment by jilladams** [comment]
Re: merging to integration branch: I think I'm now understanding that the automated tests here can't pass until the Drupal work is merged to main, yeah?

**Comment by ryguyk** [comment]
@jilladams 
>the automated tests here can't pass until the Drupal work is merged to main, yeah?
I believe this is true. In fact, it appears the tests cannot even begin because the build itself is failing. I assume this is due to trying to query data structures that are not yet in place.

**Comment by jilladams** [comment]
Ok. That in mind, repointing PR back to main, since there's no longer a use case for the integration branch.

**Review by longmd — APPROVED** [review]

**Comment by jilladams** [comment]
un-drafting / rerunning tests -- Bryan is out sick, but Drupal backend work has merged, so this PR can go ahead to review & can merge if it's in good shape.

File: src/site/stages/build/drupal/home.js [hunk 2]
```diff
@@ -70,172 +69,33 @@ function addHomeContent(contentData, files, metalsmith, buildOptions) {
...
+      const homePreviewPath = '/homepage-test';
 
-      sortedHubs = divideHubRows(sortedHubs);
+      const hero =
+        homePageHeroQuery?.itemsOfEntitySubqueueHomePageHero?.[0]?.entity || {};
+      const searchLinks = homePageOtherSearchToolsMenuQuery?.links || [];
+      const popularLinks = homePagePopularOnVaGovMenuQuery?.links || [];
+      const newsSpotlight =
+        homePageNewsSpotlightQuery
+          ?.itemsOfEntitySubqueueHomePageNewsSpotlight?.[0]?.entity || {};
```
Comment (ryguyk) [inline src/site/stages/build/drupal/home.js:89]: Very nice!

File: src/site/includes/common-tasks.drupal.liquid [hunk 3]
```diff
@@ -25,12 +25,14 @@
           </h3>
           <div class="homepage-common-tasks__search-tools">
             <ul class="vads-u-padding-left--0">
+              {% if searchLinks %}
               {% for link in searchLinks %}
               <li>
                 <i role="presentation" class="fas fa-arrow-right vads-u-color--link-default vads-u-margin-right--1"></i>
-                <a href="{{ link.link }}">{{ link.linkText }}</a>
+                <a href="{{ link.url.path }}">{{ link.label }}</a>
```
Comment (ryguyk) [inline src/site/includes/common-tasks.drupal.liquid]: I notice a very defensive approach in many places, but not all. I wonder here if `link.url.path` or `link.label` can ever be undefined or empty if `link` itself is defined. Might be worth being extra defensive.

File: src/site/includes/common-tasks.drupal.liquid [hunk 4]
```diff
@@ -44,11 +46,13 @@
             Popular on VA.gov
           </h2>
           <ul class="homepage-common-tasks__list vads-u-padding-left--0 ">
+            {% if popularLinks %}
             {% for link in popularLinks%}
             <li>
-              <a href="{{ link.link }}">{{ link.linkText}}</a>
+              <a href="{{ link.url.path }}">{{ link.label }}</a>
```
Comment (ryguyk) [inline src/site/includes/common-tasks.drupal.liquid]: Same comment about defensive checking.

File: src/site/includes/hero.drupal.liquid [hunk 5]
```diff
@@ -20,22 +20,28 @@
...
 
           <p id="myva-login--hero" class="vads-u-color--white vads-u-padding-right--5">
-            {{ copy }}
+            {% if fieldPromoText %}
+            {{ fieldPromoText }}
+            {% endif %}
           </p>
 
-          <a class="vads-c-action-link--white" href="{{ link }}">{{ linkText }}</a>
+          <a class="vads-c-action-link--white" href="{{ fieldPromoCta.entity.fieldButtonLink.url.path }}">
```
Comment (ryguyk) [inline src/site/includes/hero.drupal.liquid:34]: Same comment about defensive checks.

File: src/site/includes/news-spotlight.drupal.liquid [hunk 6]
```diff
@@ -5,8 +5,11 @@
       <!-- start first column-->
       <div
         class="vads-l-col--12 vads-u-display--none small-desktop-screen:vads-u-display--block medium-screen:vads-l-col--4">
-        <div class="homepage-blog__image demo-image-adjustments">
-          <img class="lazy" src="{{ thumbnail }}" alt="{{ thumbnailAltText }}" />
+        <div class="homepage-blog__image">
+          {% if fieldImage.entity.image.derivative.url %}
+          <img class="lazy" src="{{ fieldImage.entity.image.derivative.url }}"
+            alt="{{ fieldImage.entity.image.alt }}" />
```
Comment (ryguyk) [inline src/site/includes/news-spotlight.drupal.liquid]: Just for contrast, I'm not sure we should block rendering of the image if the alt-text property is not defined. But this is probably a question for @laflannery more than anything. If we don't have alt text provided for some reason, should we still render the image?

File: src/site/includes/news-spotlight.drupal.liquid [hunk 7]
```diff
@@ -17,13 +20,22 @@
           class="vads-u-padding--2p5 small-desktop-screen:vads-u-padding--6 small-desktop-screen:vads-u-padding-right--0 vads-u-color--white">
           <h2 class="vads-u-font-size--base vads-u-margin-top--0 vads-u-font-weight--normal vads-u-font-family--sans">VA
             NEWS</h2>
+          {% if fieldPromoHeadline %}
           <h3 class="vads-u-font-family--serif vads-u-margin-top--0 vads-u-font-size--xl">
-            <a href="{{ url }}" class="vads-u-color--white">{{ title }}</a>
+            <a href="{{ fieldLink.url.path }}" class="vads-u-color--white">{{ fieldPromoHeadline }}</a>
           </h3>
+          {% endif %}
```
Comment (ryguyk) [inline src/site/includes/news-spotlight.drupal.liquid]: Probably should check for existence of both the `href` and link text.

File: src/site/includes/news-spotlight.drupal.liquid [hunk 8]
```diff
@@ -17,13 +20,22 @@
...
                     vads-u-margin-bottom--3 small-desktop-screen:vads-u-margin-bottom--6">
-            {{ copy }} <a href="{{ url }}" class="vads-u-color--white">Read the full article</a>
+            {% if fieldPromoText %}
+            {{ fieldPromoText }}
+            {% endif %}
+            {% if fieldLinkLabel %}
+            <a href="{{ fieldLink.url.path }}" class="vads-u-color--white">
+              {{fieldLinkLabel}}
+            </a>
+            {% endif %}
```
Comment (ryguyk) [inline src/site/includes/news-spotlight.drupal.liquid:38]: Same idea. We're checking that the link text is defined, but not the link `href`.

**Review by ryguyk — CHANGES_REQUESTED** [review]

File: src/site/includes/news-spotlight.drupal.liquid (hunk 6 above)
Comment (laflannery) [reply src/site/includes/news-spotlight.drupal.liquid]: @ryguyk I do agree that we should still render the image, even if alt text is not provided.  However, the image is coming from Drupal correct? Alt text is required in the CMS, an editor should not be able to add an image without filling out the alt text field so I didn't think this scenario would apply. Is there a a different way images could be added in which alt text may not be present?

Comment (ryguyk) [reply src/site/includes/news-spotlight.drupal.liquid]: @laflannery Thanks for the feedback. You're right. It's likely not possible that the alt text would be undefined. In writing our front-end code, though, we prefer to be extra defensive in checking for existence before we attempt to render something. In this case, it doesn't seem necessary to be so defensive around the alt text.

Comment (laflannery) [reply src/site/includes/news-spotlight.drupal.liquid]: Thanks for clarifying! I just wanted to make sure I was understanding correctly.

**Review by dsasser — CHANGES_REQUESTED** [review]
Only issue I found was that the alert on the main homepage was not visible on the new homepage.
missing-alert

Other components appear to be rendering correctly, and respecting order defined in CMS:
Screen Shot 2022-11-08 at 12 06 21 PM
Screen Shot 2022-11-08 at 12 04 58 PM
Screen Shot 2022-11-08 at 12 04 01 PM
Screen Shot 2022-11-08 at 12 03 10 PM
Screen Shot 2022-11-08 at 11 57 48 AM 1

**Comment by apisandipas** [comment]
@dsasser  Great callout on the alert banner. I raised this in the Homepage Sync today and it's not a blocker at this time.

**Review by it-harrison — APPROVED** [review]

File: src/site/includes/hero.drupal.liquid (hunk 5 above)
Comment (ryguyk) [reply src/site/includes/hero.drupal.liquid:34]: Nice fixes. This is the only one I still see.

**Comment by dsasser** [comment]
> @dsasser Great callout on the alert banner. I raised this in the Homepage Sync today and it's not a blocker at this time.

Perfect, approval inbound.

**Review by dsasser — APPROVED** [review]

**Review by ryguyk — APPROVED** [review]
Thanks for fixes. Looks great.

**Pull request merged**